swim/self_evict.go

```diff
@@ -0,0 +1,244 @@
+package swim
+
+import (
+	"errors"
+	"math"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/uber/ringpop-go/util"
+
+	"github.com/uber-common/bark"
+)
+
+var (
+	// ErrDuplicateHook is returned when a hook with the given name has already
+	// been registered.
+	ErrDuplicateHook = errors.New("expected hook name to be unique")
+
+	// ErrEvictionInProgress is returned when ringpop is already in the process
+	// of evicting
+	ErrEvictionInProgress = errors.New("ringpop is already executing an eviction")
```

Comment:
nit: would it be more accurate to call this `ErrSelfEvictionInProgress` and say: "ringpop is already executing a self-eviction"?